common/src/utils/camera.rs

```diff
@@ -96,3 +102,94 @@ pub fn vr_projection_from_fov(fov: VrFov, near: f32, far: f32) -> Mat4 {
         [0.0, 0.0, a43, 0.0],
     ])
 }
+
+pub struct Orthographic {
+    screen_size: (u32, u32),
+    pub left: f32,
+    pub right: f32,
+    pub bottom: f32,
+    pub top: f32,
+    pub near: f32,
+    pub far: f32,
+    proj: [Mat4; 2],
+}
+
+
+impl Default for Orthographic{
+    fn default() -> Self {
+        Self {
+            screen_size: (1920, 1080),
+            left: -10.,
+            right: 10.,
+            bottom: -10.,
+            top: 10.,
+            near: -100.,
+            far: 100.,
+            proj: [Mat4::IDENTITY; 2],
+        }
+    }
+}
+
+impl Orthographic{
+    pub fn update_proj(&mut self, width: f32, height: f32, input: &InputEvent) {
+        
+        // Check if the screen size changes
+        if let InputEvent::Window(WindowEvent::Resized { width: screen_width, height: screen_height }) = input {
+            self.screen_size = (*screen_width, *screen_height);
+        }
+
+        // Calculate the ratio of the screen size: need to work on this part again
+        let mut x_ratio = width;
+        let mut y_ratio = height;
+
+        while x_ratio / 10. >= 1. {
```

Comment:
This section should be accomplished without a loop; e.g. with some floating point math! Would also like to see some comments explaining the math.